common/stages/Functions/answerfile

```diff
@@ -37,6 +37,14 @@ read_xml_tag()
     sed -ne 's/^.*<'"${TAG}"'\(\s\+[^>]*\)*\s*>\([^<]*\)<\/'"${TAG}"'>.*$/\2/p' <${FILE}
 }
 
+# This assumes the tag is on a single line in the file.
+sanitize_xml_tag()
+{
+    FILE="$1"
+    TAG="$2"
+    sed -i -e 's/\(^.*<'"${TAG}"'\(\s\+[^>]*\)*\s*>\)\([^<]*\)\(<\/'"${TAG}"'>.*$\)/\1SANITIZED\4/' ${FILE}
```

Comment:
This sed regex is a copy of read_xml_tag()'s sed command with additional grouping parentheses.  Those are so we can match & preserve the tag while replacing the value with the text "SANITIZED".  Since the fields are read with read_xml_tag, I wanted the sanitize routine kept in sync.  I'll take a shot at cleaning up the regex with your suggestions.

The separator doesn't require backslashes ("\") - it's sed.  Parentheses are escaped so sed does group matching.  Similarly, \s is used for whitespace matching.

We only match <$TAG once - I just tested - but it can be followed by attributes and whitespace until the closing >.  The regex could be much simpler if did not support those.